tools/azure-sdk-tools/devtools_testutils/powershell_preparer.py

```diff
@@ -52,20 +54,34 @@ def create_resource(self, name, **kwargs):
 
         if self.is_live:
             self._set_mgmt_settings_real_values()
-            for key in self.needed_keys:
+            try:
+                for key in self.needed_keys:
+
+                    scrubbed_value = self.fake_values[key]
+                    if scrubbed_value:
+                        self.real_values[key.lower()] = os.environ[key.upper()]
+                        self.test_class_instance.scrubber.register_name_pair(
+                            self.real_values[key.lower()],
+                            scrubbed_value
+                        )
+                    else:
+                        template = 'To pass a live ID you must provide the scrubbed value for recordings to \
+                            prevent secrets from being written to files. {} was not given. For example: \
+                                @PowerShellPreparer("schemaregistry", schemaregistry_endpoint="fake_endpoint.servicebus.windows.net")'
+                        raise AzureTestError(template.format(key))
+            except KeyError as key_error:
+                if not self._backup_preparers:
+                    raise key_error
```

Comment:
Just `raise`, this is re-raising the latest exception automatically. I believe this is equivalent but convention when you mostly let it go is to just write `raise`. Though I wonder if the stacktrace is impacted when you put a variable name like `key_error` in between.